### PR TITLE
[IMP] stock: move inventory at date on stock report

### PR DESCRIPTION
**Pull request by abya-odoo**
Before this commit:
==================
- Inventory at Date button was present in Locations Report.

After this commit:
==================
- Inventory at Date button is removed from Locations Report and added in Stock
Report.

task  3839744

### CODE REVIEW

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -646,6 +646,15 @@ def _count_returned_sn_products_domain(self, sn_lot, or_domains):
...
 
+    def action_inventory_at_date(self):
+        # Called when the user click on the 'Inventory at Date' button in Stock report.
+        return {
+            "res_model": "stock.quantity.history",
+            "views": [[False, "form"]],
+            "target": "new",
+            "type": "ir.actions.act_window",
+            "context": {},
+        }
```
Comment (amoyaux) [inline addons/stock/models/product.py:657]: since there is nothing dynamics, it's better to use xml

File: addons/stock/views/product_views.xml [hunk 2]
```diff
@@ -511,6 +511,9 @@
             <field name="priority" eval="100"/>
             <field name="arch" type="xml">
                 <tree sample="1" js_class="stock_report_list_view">
+                    <header>
+                        <button name="action_inventory_at_date" invisible="((context.get('inventory_mode') and not context.get('inventory_report_mode')) or context.get('no_at_date'))" string="Inventory at Date" type="object" class="btn-primary ms-1" display="always"/>
```
Comment (amoyaux) [inline addons/stock/views/product_views.xml:515]: I see the occurrence in 2 other views. Clean them too. Remove the action in python code of quant

**Comment by Whenrow** [comment]
robodoo r+

**Comment by Whenrow** [comment]
robodoo rebase-ff

**Pull request closed without merging**